LXStudio-IDE/src/main/java/titanicsend/pattern/tom/Fire.java

```diff
@@ -0,0 +1,95 @@
+package titanicsend.pattern.tom;
+
+import heronarts.lx.LX;
+import heronarts.lx.color.LXColor;
+import heronarts.lx.model.LXPoint;
+import heronarts.lx.parameter.BooleanParameter;
+import heronarts.lx.parameter.BoundedParameter;
+import heronarts.lx.parameter.CompoundParameter;
+import heronarts.lx.parameter.LXParameter;
+import titanicsend.pattern.PeriodicPattern;
+
+import static processing.core.PApplet.lerpColor;
+
+public class Fire extends PeriodicPattern {
+    private float ROW_HEIGHT = 100000;
+    private float COLUMN_WIDTH = 100000;
+    private int NUM_COLUMNS;
+    private int NUM_ROWS;
+    private int[][] buffer;
+    private int[] gradient;
+
+    protected final CompoundParameter fuel = (CompoundParameter)
+            new CompoundParameter("Fuel", 1);
+
+    public Fire(LX lx) {
+        super(lx);
+        NUM_ROWS = calculateRows();
+        NUM_COLUMNS = calculateColumns();
+        gradient = calculateGradient(LXColor.rgb(0xcf, 0x6f, 0x0f), 36);
+        buffer = new int[NUM_ROWS][NUM_COLUMNS];
+
+        register(this::spreadFire, 100);
+        addParameter("fuel", fuel);
+    }
+
+    public void spreadFire() {
+        checkFuel();
+
+        int[][] newBuffer = buffer.clone();
+        for (int row = 1; row < NUM_ROWS; row++) {
+            for (int column = 0; column < NUM_COLUMNS; column++) {
+                int rand = (int) Math.round(Math.random() * 3.0) & 3;
+                newBuffer[row][Math.min(NUM_COLUMNS - 1, Math.max(0, column - rand + 2))] = Math.max(0, buffer[row-1][column] - (1 & rand));
+            }
+        }
+
+        render();
+    }
+
+    public void render() {
+        for (LXPoint point : model.points) {
+            colors[point.index] = color(point);
+        }
+    }
+
+    private void checkFuel() {
```

Comment:
I originally defined an `onParameterChanged` method on my pattern and only updated the first row there to save some cycles when we know the value hasn't changed, but I discovered that this function isn't triggered when the parameter is changed by a modulator.